localstack/services/awslambda/provider.py

```diff
@@ -251,14 +737,1677 @@ def invoke(
+                    function_name, c.qualifier, context.account_id, context.region
+                ),
+                MaximumEventAgeInSeconds=c.maximum_event_age_in_seconds,
+                MaximumRetryAttempts=c.maximum_retry_attempts,
+                DestinationConfig=c.destination_config,
+            )
+            for c in fn.event_invoke_configs.values()
+        ]
+
+        event_invoke_configs = PaginatedList(event_invoke_configs)
+        page, token = event_invoke_configs.get_page(
+            lambda x: x,
+            marker,
+            max_items,
+        )
+        return ListFunctionEventInvokeConfigsResponse(
+            FunctionEventInvokeConfigs=page, NextMarker=token
+        )
+
+    def delete_function_event_invoke_config(
+        self, context: RequestContext, function_name: FunctionName, qualifier: Qualifier = None
+    ) -> None:
+        function_name, qualifier = api_utils.get_name_and_qualifier(
+            function_name, qualifier, context.region
+        )
+        state = lambda_stores[context.account_id][context.region]
+        fn = state.functions.get(function_name)
+        resolved_qualifier = qualifier or "$LATEST"
+        fn_arn = api_utils.qualified_lambda_arn(
+            function_name, qualifier, context.account_id, context.region
+        )
+        if not fn:
+            raise ResourceNotFoundException(
+                f"The function {fn_arn} doesn't have an EventInvokeConfig", Type="User"
+            )
+
+        config = fn.event_invoke_configs.get(resolved_qualifier)
+        if not config:
+            raise ResourceNotFoundException(
+                f"The function {fn_arn} doesn't have an EventInvokeConfig", Type="User"
+            )
+
+        del fn.event_invoke_configs[resolved_qualifier]
+
+    def update_function_event_invoke_config(
+        self,
+        context: RequestContext,
+        function_name: FunctionName,
+        qualifier: Qualifier = None,
+        maximum_retry_attempts: MaximumRetryAttempts = None,
+        maximum_event_age_in_seconds: MaximumEventAgeInSeconds = None,
+        destination_config: DestinationConfig = None,
+    ) -> FunctionEventInvokeConfig:
+        # like put but only update single fields via replace
+        state = lambda_stores[context.account_id][context.region]
+        function_name, qualifier = api_utils.get_name_and_qualifier(
+            function_name, qualifier, context.region
+        )
+
+        if (
+            maximum_event_age_in_seconds is None
+            and maximum_retry_attempts is None
+            and destination_config is None
+        ):
+            raise InvalidParameterValueException(
+                "You must specify at least one of error handling or destination setting.",
+                Type="User",
+            )
+
+        fn = state.functions.get(function_name)
+        if not fn:
+            raise ResourceNotFoundException("TODO: Where fn?")
```

Comment:
should we maybe add a user-friendly exception message, and leave the TODO in the code? ;-) ditto below